FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +20,44 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard let `self` = self else { return }
+			switch result {
+			case .success((let data, let response)):
+				self.mapSuccessResponse(data: data, response: response, completion: completion)
+			case .failure:
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+
+	private func mapSuccessResponse(data: Data, response: HTTPURLResponse, completion: @escaping (FeedLoader.Result) -> Void) {
```

Comment:
This function can be `static` since there's no need to have an instance of `RemoteFeedLoader` to perform this logic. That way, we avoid needing a strong reference to `self` inside the `load` function:
```suggestion
	private static func mapSuccessResponse(data: Data, response: HTTPURLResponse, completion: @escaping (FeedLoader.Result) -> Void) {
```

Reply:
Updated, One doubt, should we prefer `static` func over instance func for this kinda cases?

Reply:
It would depend on what's the logic being performed, and if we have any reasons to have a reference to an instance. 

If we are using instance-specific values, it might make sense to have an `instance` function, or we might prefer to pass those values as arguments in a `static` function, it would depend on the operation. 

If we are **altering** instance-specific properties, or have any type of side-effect in the instance, then I would recommend `instance` functions (instead of `inout` parameters on a function).